src/io/flutter/sdk/FlutterSdkVersion.java

```diff
@@ -132,6 +149,33 @@ public String fullVersion() {
     return version == null ? "unknown version" : version.toString();
   }
 
+  /**
+   * This considers beta versions when comparing versions. See https://github.com/flutter/flutter/wiki/Flutter-build-release-channels.
+   */
+  public int compareToWithBetaVersion(FlutterSdkVersion otherVersion) {
```

Comment:
Rename this "compareTo" and make the class extend Comparable<FlutterSdkVersion>.

It looks like this already will work for the dev channel. If you add a check for the number following "pre" it can work for all channels. My master is currently "1.26.0-2.0.pre.289", for example.